scripts/setup

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+# this shell script is responsible for:
+# - making sure your environment is setup, and
+# - setting up a database environment locally for you to use
+
+usage() {
+  echo "usage: $0 [--clean-db] [-h|--help]"
+  echo ""
+  echo "[--clean-db]     clean the database"
+  echo "[-h|--help]      print this help text"
+}
+
+# script variables and defaults
+# the list of tools required to be installed to interact with this repo
+# this is meant to verify what is also in the ./README.md#environment-setup section
+TOOLS=(java docker)
```

Comment:
Would we also want to verify the version of tools, e.g. `java -version | ...`?

Reply:
that is the "beauty" of it ... we can definitely add that if we are interested in doing so. we can be as strict or lenient as we need to be